### DISCUSSION TITLE
Deploying GluonCV models on Android

### DISCUSSION THREAD

**Discussion by LewsTherin511** [deepjavalibrary/djl#723] [Q&A]
Hi there, I need to run some GluonCV models on Android devices, and I've been usggested to use DJL.
It seems promising, since I seem to understand that with DJL I can basically use the models available in the GluonCV ModelZOO (I need to run a custom object detection, a pose estimation and an activity recognition model).

However, I'm honestly a little lost about HOW to properly do it.
I have my custom object detection model, fine-tuned from one of the pre-trained GluonCV ZOO models on a custom dataset. The model is already trained, I exported it, tested it on my computer and everything works fine.

Could you please point me towards some guides to understand how to convert/use this model on Android?

For example, I found this tutorial on [how to Load MXNet model](https://github.com/awslabs/djl/blob/master/jupyter/load_mxnet_model.ipynb)...but I assume the explained procedure allows to load MXNet models and use them within a Jupyter notebook? What would be the difference with simply using the MXNet models 'normally', i.e. without DJL?

[Here](https://github.com/aws-samples/djl-demo/tree/master/android) there's a demo on how to use a PyTorch model on Android, but there aren't many explanations on how to modify it and use other models.

**Comment by frankfliu** [comment]
The jupyter notebook is just show you how to load the model step by step, you can put them together in your java class.

We have [example for Activity recognition](https://github.com/awslabs/djl/blob/master/examples/docs/action_recognition.md) using GluonCV model.

DJL is a high level abstraction of deep learning, you can use the same java code to run inference with PyTorch/MXNet/TensorFlow model. The Android example should work for MXNet, the only thing you need is switch to MXNet engine.

Currently there is no officially MXNet android binary release, you have to download MXNet nightly build for arm and manually include in your android project. MXNet team is working on official android release, once it's done, we will publish DJL MXNet android native package to maven centeral.
